### ISSUE TITLE
[QUESTION] Why does historical_forecasts() require train_length > training_length for rolling window forecasts?

### ISSUE THREAD

**Issue by felix-ibel** [unit8co/darts#2855]
**Describe the issue linked to the documentation**
Trying to use [`historical_forecasts()`](https://github.com/unit8co/darts/blob/ec85c206f64ccf555f4e081acb5e5533359097eb/darts/models/forecasting/forecasting_model.py#L691) to generate historical forecasts with a rolling window. My model uses 20 input time steps ([`input_chunk_size=20`](https://github.com/unit8co/darts/blob/ec85c206f64ccf555f4e081acb5e5533359097eb/darts/models/forecasting/rnn_model.py#L325)) and should predict one time step, which according to the documentation results in [`training_length=21`](https://github.com/unit8co/darts/blob/ec85c206f64ccf555f4e081acb5e5533359097eb/darts/models/forecasting/rnn_model.py#L336).

I expected that this value of 21 would also be the length of each rolling window used in `historical_forecasts()`. However, setting [`train_length=21`](https://github.com/unit8co/darts/blob/ec85c206f64ccf555f4e081acb5e5533359097eb/darts/models/forecasting/forecasting_model.py#L741) is not allowed because it must be greater than or equal to [`model.min_train_series_length`](https://github.com/unit8co/darts/blob/ec85c206f64ccf555f4e081acb5e5533359097eb/darts/models/forecasting/rnn_model.py#L606).

I’m trying to understand why this additional `+1` window length is necessary in `historical_forecasts()`. From my understanding, `training_length` already defines the full input-output window required by the model, so requiring `train_length >= training_length + 1` appears unnecessary or possibly incorrect.
If there's a specific reason for this design, could it be documented more clearly?

**Additional context**
- [RNNModel Documentaion](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.rnn_model.html#recurrent-neural-networks)
- [historical_forecasts() Documentation](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.rnn_model.html#darts.models.forecasting.rnn_model.RNNModel.historical_forecasts)
- The RNNmodel Documentation of `training_length` points to `darts.utils.data.ShiftedDataset` for more Information. Unfortunately, I was unable to find documentation about `darts.utils.data.ShiftedDataset`.

**Comment by dennisbader** [comment]
Hi @felix-ibel:

- RNNModel's `training_length` allows the model to learn in an auto-regressive fashion. It is applied per batch sample (the dataset contains all possible **sub series** of your input series that have a **window size of `training_length`**). 
The `output_chunk_length` that will be used only during `model.predict()` is always `1` meaning the model will predict one step at a time. 
Having `training_length > input_chunk_length + output_chunk_length` will train the model for the auto-regressive downstream prediction / inference mode where you might also want to perform auto regression (forecast horizon > output_chunk_length). So, I recommend you to use a larger `training_length`.
- Historical forecasts' `train_length` defines how many of these RNN `training_length` **sub series** will be extracted from the target series for training (for most of the other models it would be of sub series of length `input_chunk_length + output_chunk_length`). It's the rolling training dataset window.

#### During Training:
Now regarding why `min_train_series_length = training_length + 1`: We use the `ShiftedTorchTrainingDataset` (docs are [here](https://unit8co.github.io/darts/generated_api/darts.utils.data.torch_datasets.training_dataset.html#darts.utils.data.torch_datasets.training_dataset.ShiftedTorchTrainingDataset)) with `input_chunk_length=training_length`, `output_chunk_length=training_length`, `shift=1` to train the model. The requirement for `min_train_series_length = training_length + 1` is due to the `shift=1`.

- the input chunk and output chunk have the same length `training_length`. E.g. the model produces as many predicted steps as input steps.
- the output chunk is shifted by `shift=1` so that the model learns to produce future predictions, and so that loss can be computed against the future target (shifted by 1).

#### During Prediction:
For prediction we use the `SequentialTorchInferenceDataset` (see docs [here](https://unit8co.github.io/darts/generated_api/darts.utils.data.torch_datasets.inference_dataset.html#darts.utils.data.torch_datasets.inference_dataset.SequentialTorchInferenceDataset)). The process is then:
- First let the model produce the last hidden state of the RNN using the last `input_chunk_length` points from the provided target series (note here `input_chunk_length` is the actual value that you provided at model creation, not `training_length`).
- Use the last hidden state to generate the forecasts over the entire horizon in an auto-regressive fashion. E.g. single step predictions where we feed the prediction and the corresponding last hidden state to the input for the next predicted step.

I hope this clears things up
